core/src/main/java/io/substrait/relation/ProtoRelConverter.java

```diff
@@ -137,12 +147,26 @@ private EmptyScan newEmptyScan(ReadRel rel) {
         .build();
   }
 
+  private ExtensionSingleInput newExtensionSingle(ExtensionSingleRel rel) {
+    var input = from(rel.getInput());
+    var builder =
+        ExtensionSingleInput.builder()
+            .input(input)
+            .remap(optionalRelmap(rel.getCommon()))
+            .extension(optionalAdvancedExtension(rel.getCommon()));
```

Comment:
If there was more than one type of SingleRel extension, how would those be stored? I suppose that could be encapsulated in the interpretation of the protobuf.Any data?